modules/core/src/commonMain/kotlin/fr/acinq/lightning/channel/states/Normal.kt

```diff
@@ -261,7 +261,7 @@ data class Normal(
                         }
                     }
                     is ChannelUpdate -> {
-                        if (cmd.message.shortChannelId == shortChannelId && cmd.message.isRemote(staticParams.nodeParams.nodeId, staticParams.remoteNodeId)) {
+                        if (cmd.message.isRemote(staticParams.nodeParams.nodeId, staticParams.remoteNodeId)) {
```

Comment:
I know we are already checking in `Peer`, but I think we should also call `matchesShortChannelId` here.

Reply:
I consider the channel state machine to be independent from the peer. Imo it shouldn't just take whatever is sent to it for granted. Not a huge deal

Reply:
True!